--- origsrc/gedit-plugins-2.22.2/configure.ac	2008-04-20 09:27:44.000000000 -0500
+++ src/gedit-plugins-2.22.2/configure.ac	2008-06-04 20:55:10.109375000 -0500
@@ -197,7 +197,7 @@
 EOF
 
 	if /bin/sh ../libtool --mode=compile ${CC} $PYTHON_CFLAGS -c testpython.c >/dev/null 2>&1 && \
-		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
+		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version -no-undefined $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
 		grep 'dlname.*testpython' testpython.la >/dev/null 2>&1;
 	then
 		result=yes
@@ -415,6 +415,10 @@
 fi
 
 PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+case "$host_os" in
+	cygwin*|mingw*)
+		PLUGIN_LIBTOOL_FLAGS="$PLUGIN_LIBTOOL_FLAGS -no-undefined" ;;
+esac
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 
 AC_CONFIG_FILES([
--- origsrc/gedit-plugins-2.22.2/plugins/charmap/Makefile.am	2008-03-22 08:00:26.000000000 -0500
+++ src/gedit-plugins-2.22.2/plugins/charmap/Makefile.am	2008-06-04 20:55:10.125000000 -0500
@@ -17,10 +17,8 @@
 	gedit-charmap-panel.h				\
 	gedit-charmap-panel.c
 
-libcharmap_la_LDFLAGS = \
-	$(PLUGIN_LIBTOOL_FLAGS)				\
-	$(GEDIT_LIBS)					\
-	$(GUCHARMAP_LIBS)
+libcharmap_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
+libcharmap_la_LIBADD = $(GEDIT_LIBS) $(GUCHARMAP_LIBS)
 
 # Plugin Info
 
--- origsrc/gedit-plugins-2.22.2/plugins/showtabbar/Makefile.am	2008-03-20 09:03:55.000000000 -0500
+++ src/gedit-plugins-2.22.2/plugins/showtabbar/Makefile.am	2008-06-04 20:55:10.125000000 -0500
@@ -15,6 +15,7 @@
 	gedit-show-tabbar-plugin.c
 
 libshowtabbar_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
+libshowtabbar_la_LIBADD = $(GEDIT_LIBS)
 
 # Plugin Info
 
